cmd/platform-go-challenge/main.go

```diff
@@ -0,0 +1,158 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"time"
+
+	"platform-go-challenge/config"
+	http_asset "platform-go-challenge/http/asset"
+	http_user "platform-go-challenge/http/user"
+	"platform-go-challenge/internal/app/assets"
+	"platform-go-challenge/internal/app/dashboards"
+	"platform-go-challenge/internal/app/users"
+	challenge_sql "platform-go-challenge/internal/repositories/sql"
+
+	"github.com/joho/godotenv"
+
+	"github.com/gorilla/mux"
+)
+
+func main() {
+	log.Print("app starting ...")
+	err := godotenv.Load(".env")
+	if err != nil {
+		log.Fatal("Error loading .env file")
+	}
+	cfg := config.NewConfig()
+	conn := setupConnections(cfg)
+	services := setupServices(conn)
+	r := createRouter(services, *cfg)
+	for conn.mysql.Ping() != nil {
+		log.Print("waiting for db ...")
+		time.Sleep(2 * time.Second)
+	}
+
+	up, down := DBdata()
+	defer func() {
+		for _, q := range down {
+			_, _ = conn.mysql.Exec(q)
+		}
+	}()
+	for _, q := range up {
+		_, err = conn.mysql.Exec(q)
+		if err != nil {
+			log.Print(err.Error())
+			return
+		}
+	}
+
+	_ = http.ListenAndServe(":5000", r)
+}
+
+func createRouter(services services, cfg config.Config) *mux.Router {
+	r := mux.NewRouter()
```

Comment:
Why you did not choose a web framework? Have you tested any of the most popular options?

Reply:
I figured it would be an "overkill" for a simple CRUD functionality.
Plus I felt like it would be easier for me since I haven't used a GO framework recently. I have only tried GIN 2-3 years ago.